lib/GcInfo/GcInfo.cpp

```diff
@@ -545,85 +625,71 @@ void GcInfoEmitter::encodeTrackedPointers(const GcFuncInfo *GcFuncInfo) {
+void GcInfoEmitter::encodeUntrackedPointers(const GcFuncInfo *GcFuncInfo) {
+  for (auto AllocaIterator : GcFuncInfo->AllocaMap) {
+    const AllocaInfo &AllocaInfo = AllocaIterator.second;
+    const AllocaInst *Alloca = AllocaIterator.first;
 
-  for (auto Pin : GcFuncInfo->PinnedSlots) {
-    int32_t Offset = Pin.second;
-    assert(Offset != GcInfo::InvalidPointerOffset && "Pinned Slot Not Found!");
+    assert(AllocaInfo.Offset != GcInfo::InvalidPointerOffset &&
+           "Invalid Offset for Alloca Record");
 
-    assert(SlotMap.find(Offset) == SlotMap.end() &&
-           "Pinned slot already allocated");
+    if (AllocaInfo.isGcAggregate()) {
+      encodeGcAggregate(Alloca, AllocaInfo);
+    } else if (AllocaInfo.isGcPointer()) {
+      getUntrackedSlot(AllocaInfo.Offset, AllocaInfo.isPinned());
+    }
 
-    getPinnedSlot(Offset);
-  }
-}
+    // The Followig information is not yet reported to the Runtime
+    // for the want of additional information.
+    //  -> Prolog Size
+    //  -> GS Cookie validity range
+    //
+    // TODO: The SlotOffset may need to be modified wrt Caller's SP
+    // for the following interfaces.
 
-void GcInfoEmitter::encodeSpecialSlots(const GcFuncInfo *GcFuncInfo) {
-  // The Followig information is not yet reported to the Runtime
-  // for the want of additional information.
-  //  -> Prolog Size
-  //  -> GS Cookie validity range
+    if (AllocaInfo.Flags & AllocaFlags::Alloca_GsCookie) {
+      // TODO: GC: Report GS Cookie
+      // https://github.com/dotnet/llilc/issues/768
 
-  if (GcFuncInfo->SecurityObject != nullptr) {
-// TODO: GC: Report SecurityObjectStackSlot #767
-// Encoder.SetSecurityObjectStackSlot(GcFuncInfo->SecurityObjectOffset);
-#if !defined(NDEBUG)
-    if (EmitLogs) {
-      dbgs() << "  SecurityObjectStackSlot @"
-             << GcFuncInfo->SecurityObjectOffset << " Not Reported\n";
+      // Encoder.SetGSCookieStackSlot(GcFuncInfo->GsCookieOffset,
+      //                             GcFuncInfo->GsCkValidRangeStart,
+      //                             GcFuncInfo->GsCkValidRangeEnd);
     }
-#endif // !NDEBUG
-  }
 
-  if (GcFuncInfo->GsCookie != nullptr) {
-// TODO: GC: Report GS Cookie #768
-// Encoder.SetGSCookieStackSlot(GcFuncInfo->GsCookieOffset,
-//                             GcFuncInfo->GsCkValidRangeStart,
-//                             GcFuncInfo->GsCkValidRangeEnd);
-#if !defined(NDEBUG)
-    if (EmitLogs) {
-      dbgs() << "  GSCookieStackSlot @" << GcFuncInfo->GsCookieOffset << " ["
-             << GcFuncInfo->GsCkValidRangeStart << " - "
-             << GcFuncInfo->GsCkValidRangeEnd << "]  Not Reported\n";
+    if (AllocaInfo.Flags & AllocaFlags::Alloca_GenericsContext) {
+      // TODO: GC: Report GenericsInstContextStackSlot
+      // https://github.com/dotnet/llilc/issues/766
+
+      // Encoder.SetGenericsInstContextStackSlot(
+      //    GcFuncInfo->GenericsContextOffset,
+      //    GcFuncInfo->GenericsContextParamType);
```

Comment:
I can't set this until we set GsCookie valid range correctly. Otherwise, GcInfoEncoder::Build() fails this assertion:
https://github.com/dotnet/coreclr/blob/master/src/gcinfo/gcinfoencoder.cpp#L700